swri_rospy/src/swri_rospy/__init__.py

```diff
@@ -62,3 +63,32 @@ def wrapper(*args, **kwds):
             rospy.logerr(traceback.format_exc())
             return None
     return wrapper
+
+
+def subscribe_single_threaded(name, data_class, callback=None,
```

Comment:
It's not PEP8, but I think it would be better to use CamelCase here, in the style of the `rospy.Subscriber` constructor, i.e. `swri_rospy.SingleThreadedSubscriber(...)`.

Another possibility would be to have a generator for `swri_rospy.Subscriber(...)` with an argument for `single_threaded` (which would probably default to `True`). I can imagine some things we would want to put in `swri_rospy.Subscriber()` in the future, like topic logging, and having a single entry point would make that easier.